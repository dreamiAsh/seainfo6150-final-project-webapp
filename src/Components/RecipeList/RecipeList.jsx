import React, {useEffect, useState} from "react";
import styles from "./RecipeList.module.css";
import RecipeItem from "../RecipeItem/RecipeItem"
import { isEmpty } from "lodash";

const RecipeList = (props) => {

  // const [fetchedData, setFetchedData] = useState({});
  // useEffect(() => {
  //   const fetchData = async () => {
  //     // put data fetching code here!
  //     // performs a GET request
  //     const response = await fetch(
  //       "http://demo9533681.mockable.io/recipeList"
  //     );
  //     const responseJson = await response.json();
  //     setFetchedData(responseJson);
  //   };
  //   if(isEmpty(fetchedData)) {
  //     fetchData();
  //   }
  // }, [fetchedData]);

    // const myList = Object.values(fetchedData)
    const myList = props.recipes
    const mappedList = myList.map((listItem) => (
      <RecipeItem recipe={listItem}/>
    ))

    return (
        <ul className={styles.recipeList}>{mappedList}</ul>
    )
}

export default RecipeList
