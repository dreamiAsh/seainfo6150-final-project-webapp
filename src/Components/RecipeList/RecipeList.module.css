.recipeList{
    width: 85%;
    height: 800px;
    text-align: center;
    padding: 0 7%;
    margin-bottom: 300px;
}