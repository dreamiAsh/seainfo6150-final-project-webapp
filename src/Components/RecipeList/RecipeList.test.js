import React from "react";
import { render, fireEvent } from "@testing-library/react";
import RecipeList from "./RecipeList.jsx";

it("renders the RecipeList component correctly", () => {
  const { container } = render(<RecipeList />);
  expect(container).toMatchSnapshot();
});