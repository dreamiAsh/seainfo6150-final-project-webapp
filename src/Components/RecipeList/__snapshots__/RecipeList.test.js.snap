// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the RecipeList component correctly 1`] = `
<div>
  <ul
    class="recipeList"
  />
</div>
`;
